### PR TITLE
Add percentiles utility methods.

### PR DESCRIPTION
**Pull request by rjnagal**
This is copied verbatim for our older prototype. We'll use it as
a starting point for building stats summary.

### CODE REVIEW

**Comment by vishh** [comment]
LGTM

File: utils/percentiles.go [hunk 1]
```diff
@@ -0,0 +1,126 @@
...
+	// Average over the collected sample.
+	Mean uint64 `json:"mean"`
+	// Max seen over the collected sample.
+	Max uint64 `json:"max"`
+	// 90th percentile over the collected sample.
+	Ninety uint64 `json:"ninety"`
+}
+
+// Get 90th percentile of the provided samples. Round to integer.
+func Get90Percentile(samples uint64Slice) uint64 {
```
Comment (vmarmol) [inline utils/percentiles.go]: what about making it a method on the object (uint64Slice)?

File: utils/percentiles.go [hunk 2]
```diff
@@ -0,0 +1,126 @@
...
+	index := int(idx)
+	percentile := float64(samples[index-1])
+	if index > 1 || index < count {
+		percentile += frac * float64(samples[index]-samples[index-1])
+	}
+	return uint64(percentile)
+}
+
+// Add new sample to existing average.
+func GetMean(mean float64, value uint64, count uint64) float64 {
```
Comment (vmarmol) [inline utils/percentiles.go]: WDYT:

func NewMean(size uint64) {
  // .. create ...
}

type Mean struct {
  size uint64
  mean uint64
}

func (self *Mean) Add(value uint64) {
  // Do what you did here
}

File: utils/percentiles.go [hunk 3]
```diff
@@ -0,0 +1,126 @@
...
+	if count < 1 {
+		return 0
+	}
+	c := float64(count)
+	v := float64(value)
+	return (mean*(c-1) + v) / c
+}
+
+func GetPercentiles(stats []*info.ContainerStats) (Percentiles, Percentiles) {
```
Comment (vmarmol) [inline utils/percentiles.go]: Can we add a comment saying what output is what?

Comment (vmarmol) [inline utils/percentiles.go]: And this is memory usage and total CPU usage

Comment (vmarmol) [inline utils/percentiles.go]: Ah, it's working set :)

File: utils/percentiles.go (hunk 1 above)
Comment (rjnagal) [inline utils/percentiles.go]: Made it a method.

File: utils/percentiles.go (hunk 2 above)
Comment (rjnagal) [inline utils/percentiles.go]: Yeah, that's nicer. Changed :)

File: utils/percentiles.go (hunk 3 above)
Comment (rjnagal) [inline utils/percentiles.go]: Updated comment.

**Comment by vmarmol** [comment]
LGTM

**Pull request merged**